fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/ModifyInference.java

```diff
@@ -0,0 +1,181 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.optimizer.rule.mv;
+
+import com.starrocks.catalog.KeysType;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.Table;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptExpressionVisitor;
+import com.starrocks.sql.optimizer.operator.physical.PhysicalOlapScanOperator;
+import com.starrocks.sql.optimizer.operator.physical.stream.PhysicalStreamAggOperator;
+import com.starrocks.sql.optimizer.operator.physical.stream.PhysicalStreamJoinOperator;
+import com.starrocks.sql.optimizer.operator.physical.stream.PhysicalStreamScanOperator;
+import org.apache.commons.lang3.NotImplementedException;
+
+import java.util.EnumSet;
+import java.util.Objects;
+
+/**
+ * Infer the ModifyOp for operator, include INSERT/UPDATE/DELETE
+ */
+public class ModifyInference extends OptExpressionVisitor<ModifyInference.ModifyOp, Void> {
+
+    private static final ModifyInference INSTANCE = new ModifyInference();
+
+    public static ModifyOp infer(OptExpression optExpr) {
+        return optExpr.getOp().accept(INSTANCE, optExpr, null);
+    }
+
+    @Override
+    public ModifyOp visit(OptExpression optExpression, Void ctx) {
+        throw new NotImplementedException("Operator not supported: " + optExpression);
+    }
+
+    @Override
+    public ModifyOp visitPhysicalFilter(OptExpression optExpression, Void ctx) {
+        return visit(optExpression.inputAt(0), ctx);
+    }
+
+    @Override
+    public ModifyOp visitPhysicalProject(OptExpression optExpression, Void ctx) {
+        return visit(optExpression.inputAt(0), ctx);
+    }
+
+    // TODO(murphy) read from user property, support custom the ModifyOp behavior
+    private ModifyOp visitOlapTable(Table table) {
+        if (!table.isOlapTable()) {
+            throw new NotImplementedException("Only OLAP table is supported");
+        }
+        OlapTable olapTable = (OlapTable) table;
+        ModifyOp res;
+        if (olapTable.getKeysType().equals(KeysType.DUP_KEYS)) {
+            res = ModifyOp.INSERT_ONLY;
```

Comment:
we could delete the DUP_KEYS table.

Reply:
That's a logical delete, which could not handled by mv right now. The stream operator could only accept the physical delete